typing/ctype.ml

```diff
@@ -2420,21 +2458,20 @@ let rec estimate_type_jkind ~expand_component ~ignore_mod_bounds env ty =
   | Tarrow _ -> Jkind.for_arrow
   | Ttuple elts -> Jkind.for_boxed_tuple elts
   | Tunboxed_tuple ltys ->
-     let tys_modalities =
-       List.map
-         (fun (_lbl, ty) ->
-            let { ty; modality } = expand_component ty in
-            (ty, modality))
-         ltys
-     in
+     let components = List.map (fun (_lbl, ty) -> expand_component ty) ltys in
      (* CR layouts v2.8: This pretty ridiculous use of [estimate_type_jkind]
         just to throw most of it away will go away once we get [layout_of].
         Internal ticket 2912. *)
      let layouts =
-       List.map (fun (ty, _modality (* ignore; we just care about layout *)) ->
-         estimate_type_jkind ~expand_component ~ignore_mod_bounds env ty |>
-         Jkind.extract_layout)
-         tys_modalities
+       List.map (fun unwrapped_ty ->
```

Comment:
Oof.

Maybe break `rewrap_jkind_l` into `rewrap_layout_l` and then just use that piece? This is pretty sad.

And it's also a bit sad to do three `List.map`s in a row. This isn't Haskell -- each intermediate list actually gets built!